server/monitor-types/nut.js

```diff
@@ -0,0 +1,64 @@
+const { MonitorType } = require("./monitor-type");
+const { UP, DOWN } = require("../../src/util");
+const dayjs = require("dayjs");
+const jsonata = require("jsonata");
+const Nut = require("node-nut");
+
+const { log } = require("../../src/util");
+
+class NutMonitorType extends MonitorType {
+
+    name = "nut";
+
+    /**
+     * @inheritdoc
+     */
+    async check(monitor, heartbeat, _server) {
+        let startTime = dayjs().valueOf();
+        let expression = jsonata(monitor.jsonPath);
+
+        const nut = new Nut(monitor.port, monitor.hostname);
+
+        nut.on("ready", () => {
+            nut.GetUPSList((upslist, err) => {
+                if (err) {
+                    nut.close();
+                    log.error("NUT Error: " + err);
+                }
+
+                let upsname = upslist[monitor.upsName] && monitor.upsName || Object.keys(upslist)[0];
+
+                nut.GetUPSVars(upsname, async (vars, err) => {
+                    nut.close();
+                    if (err) {
+                        throw new Error("Error getting UPS variables");
+                    } else {
+                        // convert data to object
```

Comment:
Could you clarify that `data` can be a json-string or object according to the usage?